test/vcf/record_cache_test.cpp

```diff
@@ -142,8 +153,43 @@ namespace ebi
         SECTION("Symbolic Duplicates")
         {
             CHECK( count_duplicates_and_rethrow_error(cache, {107, "C", {"<INS>"}}) == 0 );
-            // This won't throw an error. Only warning would be generated, which isn't detected here.
+            // This won't throw an error.
             CHECK( count_duplicates_and_rethrow_error(cache, {107, "C", {"<INS>"}}) == 0 );
         }
     }
+
+    TEST_CASE("Symbolic Duplicates test")
+    {
+        vcf::RecordCache cache{0};
+
+        cache.check_duplicates(build_mock_record({100, "A", {"T"}}));
+        cache.check_duplicates(build_mock_record({101, "A", {"T"}}));
+        cache.check_duplicates(build_mock_record({102, "A", {"T"}}));
+        cache.check_duplicates(build_mock_record({103, "A", {"T"}}));
+        cache.check_duplicates(build_mock_record({104, "A", {"T"}}));
+        cache.check_duplicates(build_mock_record({105, "A", {"T"}}));
+        cache.check_duplicates(build_mock_record({107, "C", {"<INS>"}}));
```

Comment:
half of these are not relevant. keep only those that allow interesting test checks